src/plugins.c

```diff
@@ -747,32 +876,84 @@ plugin_cleanup(Plugin *plugin)
 		plugin->cb_data_destroy = NULL;
 	}
 
+	proxied_count_dec(plugin->proxy);
 	geany_debug("Unloaded: %s", plugin->filename);
 }
 
 
+/* Remove all plugins that proxy is a proxy for from plugin_list (and free) */
+static void free_subplugins(Plugin *proxy)
+{
+	GList *item;
+
+	item = plugin_list;
+	while (item)
+	{
+		GList *next = g_list_next(item);
+		if (proxy == ((Plugin *) item->data)->proxy)
+		{
+			/* plugin_free modifies plugin_list */
+			plugin_free((Plugin *) item->data);
+		}
+		item = next;
+	}
+}
+
+
+/* Returns true if the removal was successful (=> never for non-proxies) */
+static gboolean unregister_proxy(Plugin *proxy)
+{
+	gboolean is_proxy = FALSE;
+	GList *node;
+
+	/* Remove the proxy from the proxy list first. It might appear more than once (once
+	 * for each extension), but if it doesn't appear at all it's not actually a proxy */
+	foreach_list_safe(node, active_proxies.head)
```

Comment:
prev function does the exact same thing without a macro.  it would be more consistent using the same technique in the 2 places (either one is OK with me, @codebrainz would prefer previous func's one)